prov/gni/src/gnix_vc.c

```diff
@@ -526,6 +537,7 @@ static int __gnix_vc_connect_to_same_cm_nic(struct gnix_vc *vc)
 		   vc_peer);
 
 exit_w_lock:
+	vc->peer_caps = ep->caps;
```

Comment:
I think this should not be done here, since `exit_w_lock` is primarily an error exit label and I don't think we should modify the vc unless there are no errors.  I think this line can move up above the label, and also inserted after line 449 (the only non-error exit to this label).